### PR TITLE
Improve configuration

### LINKED ISSUE #5: Use HttpClient in ApiClient constructor
I would create an `ApiClient::configure` method which accepts the current parameters and injects the HTTP client. Better in case someone configures a client himself.

### PR DESCRIPTION
**Pull request by sagikazarmark**
Closes #5

### CODE REVIEW

File: src/ApiClient.php [hunk 1]
```diff
@@ -34,27 +34,38 @@ class ApiClient
     private $requestBuilder;
 
     /**
```
Comment (Nyholm) [inline src/ApiClient.php:35]: We should add comment about using `configure`

**Review by Nyholm — COMMENTED** [review]
Im not too happy with that this is the simplest way to create a new API client with an API key:

```php
$configurator = new HttpClientConfigurator();
$configurator->setApiKey('key');
$client = Client::configure($configurator);
```

What do you thing about **adding**: 

```php
$client = Client::createByApiKey('key');
```

---------------

Apart form the above, I like the PR. But the `configure` factory class does not do too much but running one line of code... Hm.. maybe we should make it more useful to cover the 80% of the cases when constructing the Client

**Comment by Nyholm** [comment]
Just FYI: We should also be aware of how to use the container injection. https://github.com/php-translation/platform-adapter/blob/master/src/Loco/src/Bridge/Symfony/DependencyInjection/TranslationAdapterLocoExtension.php#L43

**Comment by sagikazarmark** [comment]
I think we can just use the factory method, no?

**Review by Nyholm — APPROVED** [review]

**Pull request merged**